src/Gui/ToolHandler.h

```diff
@@ -53,6 +53,9 @@ class GuiExport ToolHandler
     /// enabling to set data member dependent icons (i.e. for different construction methods)
     void updateCursor();
 
+    virtual std::list<InputHint> getToolHints() const;
```

Comment:
Could it be relevant to define in .h instead of .cpp?
```suggestion
    virtual std::list<InputHint> getToolHints() const { return {} };
```